services/matrix/matrix.service.js

```diff
@@ -64,22 +90,51 @@ module.exports = class Matrix extends BaseJsonService {
     })
   }
 
-  async fetch({ host, roomId }) {
-    let auth
-    try {
-      auth = await this.registerAccount({ host, guest: true })
-    } catch (e) {
-      if (e.prettyMessage === 'guests not allowed') {
-        // attempt fallback method
-        auth = await this.registerAccount({ host, guest: false })
-      } else throw e
+  async lookupRoomAlias({ host, roomAlias, accessToken }) {
+    return this._requestJson({
+      url: `https://${host}/_matrix/client/r0/directory/room/%23${roomAlias}`,
+      schema: matrixAliasLookupSchema,
+      options: {
+        qs: {
+          access_token: accessToken,
+        },
+      },
+      errorMessages: {
+        401: 'bad auth token',
+        404: 'room not found',
+        429: 'rate limited by rooms host',
+      },
+    })
+  }
+
+  async fetch({ roomAlias, serverFQDN }) {
+    let host
+    if (serverFQDN === undefined) {
+      const splitAlias = roomAlias.split(':')
+      // A room alias can either be in the form #localpart:server or
+      // #localpart:server:port. In the latter case, it's wiser to skip the name
+      // resolution and use that value right away.
+      switch (splitAlias.length) {
+        case 2:
+          host = splitAlias[1]
+          break
+        case 3:
+          host = splitAlias[2] + splitAlias[3]
+          break
+        default:
+          throw new errors.InvalidParameter()
```

Comment:
Will this be immediately obvious to users what the error is with the value they have provided for `roomAlias` or would it be helpful to add a `prettyMessage`? For example:

```js
...InvalidParameter({ prettyMessage: ' whatever makes sense here' })
```

Reply:
Hah, right, that was before the second (optional) parameter was added, so it was obvious back then, but not that much now.

Reply:
Fixed in https://github.com/badges/shields/pull/2527/commits/75b799d01f4d21914125d75db7a17d5887384161